### PR TITLE
[ENH] Itakura parallelogram bounding

### PR DESCRIPTION
**Pull request by chrisholder**
#### What does this implement/fix? Explain your changes.
This PR adds in a new bounding matrix technique called Itakura parallelogram. This bounding constructs a parallogram on the matrix which means there is less warping enabled at the start and end of the path and full warping allowed in the middle. 

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### Did you add any tests for the change?
Yes and updated all the distances tests to ensure it is working for them all.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.
- [x] For new estimators, I've added the estimator to the online documentation.

### CODE REVIEW

File: aeon/distances/_ddtw.py [hunk 1]
```diff
@@ -149,6 +170,9 @@ def ddtw_pairwise_distance(
     window: float, default=None
         The window to use for the bounding matrix. If None, no bounding matrix
         is used.
+    itakura_max_slope: float, defaults=None
```
Comment (TonyBagnall) [inline aeon/distances/_ddtw.py:173]: fix formatting, e.g.
itakura_max_slope : float, default=None

File: aeon/distances/_ddtw.py [hunk 2]
```diff
@@ -149,6 +170,9 @@ def ddtw_pairwise_distance(
...
+        Maximum slope as a % of the number of time points used to create Itakura
+        parallelogram on the bounding matrix. Must be between 0. and 1..
```
Comment (TonyBagnall) [inline aeon/distances/_ddtw.py:175]: Must be between 0 and 1.

File: aeon/distances/_ddtw.py [hunk 3]
```diff
@@ -222,12 +250,14 @@ def _ddtw_pairwise_distance(X: np.ndarray, window: float) -> np.ndarray:
 
 @njit(cache=True, fastmath=True)
 def _ddtw_from_multiple_to_multiple_distance(
-    x: np.ndarray, y: np.ndarray, window: float
+    x: np.ndarray, y: np.ndarray, window: float, itakura_max_slope: float = None
```
Comment (TonyBagnall) [inline aeon/distances/_ddtw.py:253]: should this not have a default?

File: aeon/distances/_ddtw.py [hunk 4]
```diff
@@ -259,6 +289,9 @@ def ddtw_alignment_path(
     window: float, default=None
         The window to use for the bounding matrix. If None, no bounding matrix
         is used.
+    itakura_max_slope: float, defaults=None
```
Comment (TonyBagnall) [inline aeon/distances/_ddtw.py:292]: itakura_max_slope : float, default=None

**Review by TonyBagnall — CHANGES_REQUESTED** [review]

File: aeon/distances/_lcss.py [hunk 5]
```diff
@@ -69,6 +73,9 @@ def lcss_distance(
     epsilon : float, default=1.
         Matching threshold to determine if two subsequences are considered close
         enough to be considered 'common'. The default is 1.
+    itakura_max_slope : float, default=None
+        Maximum slope as a % of the number of time points used to create Itakura
```
Comment (TonyBagnall) [inline aeon/distances/_lcss.py:77]: proportion not a percentage

**Review by TonyBagnall — DISMISSED** [review]

File: aeon/distances/_lcss.py (hunk 5 above)
Comment (chrisholder) [reply aeon/distances/_lcss.py:77]: Right ill change that everywhere

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**